src/common/service/acl.ts

```diff
@@ -0,0 +1,98 @@
+import { Injectable, Inject, forwardRef } from '@nestjs/common';
+import { PrismaService } from '../../prisma.service';
+import { BenefitsService } from '../../benefits/benefits.service';
+
+@Injectable()
+export class AclService {
+    constructor(
+        private readonly prisma: PrismaService,
+        @Inject(forwardRef(() => BenefitsService))
+        private readonly benefitsService: BenefitsService
+    ) {}
+
+    /**
+     * Common function to fetch benefit data
+     * @param benefitId - The ID of the benefit
+     * @returns Object containing benefit data
+     */
+    private async getBenefitData(benefitId: string, authToken: string, userId: number): Promise<boolean> {
```

Comment:
@milindSansare the function is not only fetching the benefit, it is also checking if the user has access to the benefit.
the funciton name should be `getBenefitData` => `canAccessBenefit`

also the function `canAccessBenefit` is calling the above function and returning the response as it is

you can remove existing `canAccessBenefit` function and rename this function to `canAccessBenefit` and use the same in the `canAccessApplication` function